### DISCUSSION TITLE
Can't launch transmission on a Synology using NORDVPN provider

### DISCUSSION THREAD

**Discussion by LoyerG** [haugene/docker-transmission-openvpn#1993] [Q&A] [answered]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Config used

Trying to install transmission open-VPN on a NAS synologu DS220+ using portainer. I use NORDVPN as a provider

### Current Behavior

It seems that it doesn't want to start, don't know why

### Expected Behavior

Launching transmission with NORDVPN's VPN active

### How have you tried to solve the problem?

This is my docker compose: 

```
`version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        devices:
            - /dev/net/tun
        volumes:
          - /volume1/docker/transmission-openvpn:/data
          - /volume1/downloads:/data/completed
          - /volume1/docker/transmission-openvpn/resolv.conf:/etc/resolv.conf
        environment:
          - PUID=1026
          - PGID=100
          - CREATE_TUN_DEVICE=true
          - OPENVPN_PROVIDER=NORDVPN
          - OPENVPN_USERNAME=*****
          - OPENVPN_PASSWORD=*****
          - WEBPROXY_ENABLED=false
          - NORDVPN_COUNTRY=US
          - NORDVPN_PROTOCOL=tcp
          - NORDVPN_CATEGORY=legacy_p2p
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn
```

### Log output
```

Starting container with revision: ae45b1f22178ddee089d9ea10d4732dbde2d8ef2,
WARNING: initial DNS resolution test failed,
Creating TUN device /dev/net/tun,
mknod: /dev/net/tun: File exists,
Using OpenVPN provider: NORDVPN,
Running with VPN_CONFIG_SOURCE auto,
Provider NORDVPN has a bundled setup script. Defaulting to internal config,
Executing setup script for NORDVPN,
2021-10-07 08:34:09 Checking curl installation,
2021-10-07 08:34:09 Removing existing configs,
2021-10-07 08:34:09 Selecting the best server...,
2021-10-07 08:34:09 Searching for technology: openvpn_tcp,
2021-10-07 08:34:09 Unable to find a server with the specified parameters, using any recommended server,
2021-10-07 08:34:09 Best server : ,
2021-10-07 08:34:09 Downloading config: default.ovpn,
2021-10-07 08:34:09 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/.tcp.ovpn,
curl: (6) Could not resolve host: downloads.nordcdn.com,
No VPN configuration provided. Using default.,
Modifying /etc/openvpn/nordvpn/default.ovpn for best behaviour in this container,
Modification: Point auth-user-pass option to the username/password file,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Change ca certificate path,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Change ping options,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Update/set resolv-retry to 15 seconds,
Modification: Change tls-crypt keyfile path,
Modification: Set output verbosity to 3,
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop,
Setting OpenVPN credentials...,
Thu Oct  7 08:34:49 2021 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode,
Options error: You must define TUN/TAP device (--dev),
Use --help for more information.
```

### Environment

```markdown
- OS: DSM 7.0
- Docker: 20.10.3-1233
```

**Comment by pkishino** [comment]
Please read the pinned issue regarding DSM.
Your problem is DNS not resolving, second line..
Why are you mounting in resolv.conf? You should check the content, probably wrong dns…this is a config problem

**Comment by LoyerG** [comment]
OK so, I simplified everything, trying not to use the resolv.conf file. Here is my docker-compose: 

```
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        devices:
            - /dev/net/tun
        volumes:
            - /volume1/docker/transmissionVPN:/data
            - /volume1/downloads:/data/completed
        environment:
            - OPENVPN_PROVIDER=NORDVPN
            - OPENVPN_USERNAME=***
            - OPENVPN_PASSWORD=***
            - NORDVPN_COUNTRY=US
            - NORDVPN_PROTOCOL=tcp
            - NORDVPN_CATEGORY=legacy_p2p
            - LOCAL_NETWORK=192.168.1.1/16
            - PUID=1026
            - PGID=100
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - 9091:9091
        image: haugene/transmission-openvpn
```
 And here are the log: 

```
Starting container with revision: ae45b1f22178ddee089d9ea10d4732dbde2d8ef2,
WARNING: initial DNS resolution test failed,
Creating TUN device /dev/net/tun,
mknod: /dev/net/tun: File exists,
Using OpenVPN provider: NORDVPN,
Running with VPN_CONFIG_SOURCE auto,
Provider NORDVPN has a bundled setup script. Defaulting to internal config,
Executing setup script for NORDVPN,
2021-10-08 07:02:09 Checking curl installation,
2021-10-08 07:02:09 Removing existing configs,
2021-10-08 07:02:09 Selecting the best server...,
2021-10-08 07:02:09 Searching for technology: openvpn_tcp,
2021-10-08 07:02:09 Unable to find a server with the specified parameters, using any recommended server,
2021-10-08 07:02:09 Best server : ,
2021-10-08 07:02:09 Downloading config: default.ovpn,
2021-10-08 07:02:09 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/.tcp.ovpn,
curl: (6) Could not resolve host: downloads.nordcdn.com,
No VPN configuration provided. Using default.,
Modifying /etc/openvpn/nordvpn/default.ovpn for best behaviour in this container,
Modification: Point auth-user-pass option to the username/password file,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Change ca certificate path,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Change ping options,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
sed: can't read /etc/openvpn/nordvpn/default.ovpn: No such file or directory,
Modification: Update/set resolv-retry to 15 seconds,
Modification: Change tls-crypt keyfile path,
Modification: Set output verbosity to 3,
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop,
Setting OpenVPN credentials...,
adding route to local network 192.168.1.1/16 via 172.22.0.1 dev eth0,
RTNETLINK answers: Invalid argument,
Fri Oct  8 07:02:49 2021 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode,
Options error: You must define TUN/TAP device (--dev),
Use --help for more information.
```

I have the same issue with the DNS, but I didn't touch anything. I really don't get it. I read the pinned issue regarding DNS but it's not the same problem I encounter.

**Comment by pkishino** [comment]
Well, config looks fine but the container doesn’t have a connection at all
it seems. Please look into your docker setup/host.. does it resolve dns
properly? Try another barebones container? Before the tunnel is up it
should be using direct host dns to resolve if you don’t override this.. so
looks more like something not correct with docker

**Reply by LoyerG** [reply]
I use other containers as jackett, radarr or even plex and I don't have any problem with them. I'll check my docker config then

**Comment by pkishino** [comment]
Hmm, are you using any other network config with those containers?

**Reply by LoyerG** [reply]
Nope. I use the bridge connection

**Reply by pkishino** [reply]
As you see here:
if ! nslookup ${HEALTH_CHECK_HOST:-"google.com"} 1>/dev/null 2>&1; then

In your case, as you don’t use Dns override nothing has been changed, but dns lookup falls here, so something problematic with docker.. are you setting dns for those other containers?

**Reply by pkishino** [reply]
any progress?

**Comment by LoyerG** [comment]
No nothing. i checked everything I don't understand why the DNS fails. So
now I just use regular transmission and I plug my syno on the VPN. Thx for
your help anyway





Le jeu. 14 oct. 2021 à 02:39, Patrick Kishino ***@***.***> a
écrit :

**Reply by pkishino** [reply]
You might want to try and create a dedicated network instead of bridge network..
Are you perhaps using skynet or pihole etc where dns might bounce or so? You could try and troubleshoot from inside the container by overriding /etc/resolv.conf and adding nameserver 8.8.8.8 to it and see if dns will resolve

**Comment by LoyerG** [comment]
I could do that indeed. I'm trying this as soon as I have time to start
over. Thanks a lot for your help


Cordialement,

Grégory LOYER
| mobile: 06.34.68.14.39
| email: ***@***.***



Le jeu. 14 oct. 2021 à 11:44, Patrick Kishino ***@***.***> a
écrit :

**Comment by GKHerping** [comment, marked as answer]
Hello

To get my DSM Docker to work I've had to do the following:

1. Add a file called `resolv.conf` at my /volume1/docker/*custom_folder*/ folder. Below is what I have written in the .conf
```
nameserver 8.8.8.8
nameserver 8.8.4.4
nameserver 1.1.1.1
```
2. Go to DSM - Docker - Container - transmission-openvpn (container name) - Edit - Volume
3. Add File - Add the above file. 
4. Set Mount path to: /etc/resolv.conf
5. Execute **without** high privilege
6. It fails
7. Execute **with** high privilege
8. It works

**Reply by pkishino** [reply]
I did read that latest dsm update should have fixed some issues with docker.. also, I don't recommend using the DSM docker app

**Reply by DoubleDutch87** [reply]
Experiencing a similar problem. Did what you wrote, but it still fails with the following error:

Starting container with revision: 76f7a2ce2957c1b48d5dee0c713a3d15903a5905
WARNING: initial DNS resolution test failed
Creating TUN device /dev/net/tun
mknod: /dev/net/tun: File exists
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
Downloading user specified config. NORDVPN_PROTOCOL is set to: UDP
2021-12-10 15:21:59 Checking curl installation
2021-12-10 15:21:59 Removing existing configs
2021-12-10 15:21:59 Selecting the best server...
2021-12-10 15:21:59 Searching for group: legacy_p2p
2021-12-10 15:21:59 Searching for technology: openvpn_udp
2021-12-10 15:21:59 Best server : <name>.nordvpn.com
2021-12-10 15:21:59 Downloading config: default.ovpn
2021-12-10 15:21:59 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/nl885.nordvpn.com.udp.ovpn
2021-12-10 15:21:59 Using OpenVPN CONFIG :: hu
2021-12-10 15:21:59 Downloading config: hu.ovpn
2021-12-10 15:21:59 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/hu.udp.ovpn
Starting OpenVPN using config hu.ovpn
Modifying /etc/openvpn/nordvpn/hu.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 192.168.1.0/16 via 172.19.0.1 dev eth0
RTNETLINK answers: Invalid argument
Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/nordvpn/hu.ovpn:1: html (2.4.7)
